native/cocos/renderer/pipeline/RenderFlow.cpp

```diff
@@ -51,33 +51,27 @@ void RenderFlow::activate(RenderPipeline *pipeline) {
         return s1->getPriority() < s2->getPriority();
     });
 
-    for (auto *const stage : _stages) {
+    for (auto const &stage : _stages) {
         stage->activate(pipeline, this);
     }
 }
 
 void RenderFlow::render(scene::Camera *camera) {
-    for (auto *const stage : _stages) {
+    for (auto const &stage : _stages) {
         stage->render(camera);
     }
 }
 
 void RenderFlow::destroy() {
-    if (_isResourceOwner) {
-        for (auto *stage : _stages) {
-            CC_SAFE_DESTROY_AND_DELETE(stage);
-        }
-    } else {
-        for (auto *stage : _stages) {
-            CC_SAFE_DESTROY(stage);
-        }
+    for (auto &stage : _stages) {
+        CC_SAFE_DESTROY(stage);
     }
```

Comment:
In C++, i think it is better to invoke destroy in destructor, then don't need to destroy here. Even better is that, don't need destroy function.

Reply:
The reason is that pipeline contains gfx resources which need to be destroyed immediately instead of GC. 
`destroy` is dispatched from `root.destroy->Root::destroy->RenderPipeline.cpp->RenderFlow.cpp`. So the timing of destroy might be several seconds before `pipeline` is garbage collected.
We will try to provide a better solution to the resource life-cycle problem, in the new pipeline.

Reply:
Ok.